test/html/pipeline/mention_filter_test.rb

```diff
@@ -72,6 +72,34 @@ def test_links_to_more_info_when_info_url_given
       filter(body, '/', 'https://github.com/blog/821').to_html
   end
 
+  def test_base_url_slash
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '/').to_html
+  end
+
+  def test_base_url_under_custom_route
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/userprofile/jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '/userprofile').to_html
+  end
+
+  def test_base_url_tilde
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/~jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '~').to_html
+  end
```

Comment:
@JuanitoFatas Thanks for the updated tests.  :smile: 

They look good expect `test_base_url_tilde`.  It's not our responsibility to guard against malformed [`base_url`](https://github.com/JuanitoFatas/html-pipeline/blob/fix/mention-filter-tilde/lib/html/pipeline/filter.rb#L97-L101) input.  If we remove this test, our implementation is [greatly simplified](https://github.com/jch/html-pipeline/commit/4b27f93c5c2679840847957a2935f85d5d5a385b) [(CI Run)](https://travis-ci.org/jch/html-pipeline/builds/46010905).  

Thoughts?